src/tableView.js

```diff
@@ -411,7 +411,17 @@ class TableView {
    * @returns {number}
    */
   countRenderableColumns() {
-    return Math.min(this.instance.columnIndexMapper.getRenderableIndexesLength(), this.settings.maxCols);
+    const numberOfColumns = Math.min(
+      this.instance.columnIndexMapper.getNotTrimmedIndexesLength(), this.settings.maxCols);
+    // Don't take hidden columns into account.
+    const firstNotHiddenColumn = this.instance.columnIndexMapper.getFirstNotHiddenIndex(
+      numberOfColumns - 1, -1);
+
+    if (firstNotHiddenColumn === null) {
+      return 0;
+    }
+
+    return this.instance.columnIndexMapper.getRenderableFromVisualIndex(firstNotHiddenColumn) + 1;
```

Comment:
I understand the mechanism and it's clear to read. But I wonder if's is possible to not duplicate the code between these two methods? Maybe create a util method that will count columns or rows based on the provided argument?

Reply:
Thanks for the suggestion. I did it within: https://github.com/handsontable/handsontable/pull/7471/commits/0a77b75365e31ca799d09245058d7efe1ce78bd3. Surprisingly, it does not look weird as I expected 😃